main.cpp

```diff
@@ -1,475 +1,581 @@
-    3)
- */
-
-/*
-Part 1d
-You now know how to define a UDT that is composed of other UDT.
-Now you will learn how to break down those sub-object UDTs into 
-their 5 properties and 3 actions.
-
-The goal of Part 1d is to get you to think about breaking down an 
-object into smaller and smaller objects, until the smallest object 
-is made of up only C++ primitives and std::string. 
-
-Revisiting the previous example:  
-Cell Phone
-
-A Cell Phone is made up of the following 5 properties/sub-objects 
-and 3 actions:
-    Display
-    Memory
-    CPU
-    Radio
-    Applications
-3 actions:
-    make a call
-    send a text
-    run an application.
-
-These 5 properties can be broken down into their own sub-objects 
-and properties. 
-
-If we break down the first property 'Display' into its 5 properties 
-we get:
-    brightness
-    amount of power consumed.
-    pixels
-    width in cm
-    height in cm
-
-the Display's brightness can be represented with a Primitive, 
-such as a double. 
-
-The amount of power consumed can also be represented with a 
-Primitive, such as a float or integer (i.e. 250mWa)
-
-The 'pixels' property must be represented with an array of Pixel 
-instances, as the screen has more than 1 row of pixels.
-    Arrays have not been discussed and can't be used in this 
-    project.
-    Instead, we can use an Integer primitive to store the 
-    Number of Pixels:
-
-Display:
-    Number of Pixels
-    Amount of Power consumed (milliwatt-hours)
-    Brightness
-    width in cm
-    height in cm
-
-As you can see, the 'Display' UDT has been broken down to the 
-point of being able to describe it with C++ primitives. 
+    1) host local artists work on walls
+    2) charge customer
+    3) train staff
 */
 
+struct CoffeeShop
+{
+
+    //number of chairs (int)
+    int numChairs = 54;
+    //number of staff (int)
+    int numStaff = 7;
+    //number of toilets (int)
+    int numToilets = 2;
+    //annual profit (double)
+    double annualProfit = 45362.78;
+    //price of coffee (float)
+    float priceCoffee = 3.40f;
+
+    struct Employee
+    {
+        std::string firstName = "Magnus";
+        std::string lastName = "Magnusson";
+        int employeeID = 12345;
+        bool fullyTrained = true;
+        float salary = 30000.00f;
+
+        void makeDeliveries();
+        void takeHoliday();
+        bool deservesPromotion(); // returns whether the employee is derving of a promotion
+
+
+
+    };
+
+    //host local artists work on walls
+    float getArtExhibitCommision(float percenatgeCommisionAgreed, float totalArtSales, CoffeeShop coffeeshop); //returns the total value of commission earn from the sale of all the art at this instance of CoffeeShop 
+    //charge customer
+    float getCustomerSale(float priceCoffee, int numberOfCoffees, int customerNumber); //returns the total value of the sale of all good for this customer
+    //train staff
+    void staffTraining(Employee employee, std::string trainingRequired);
```

Comment:
Your function names do not match the plain-english descriptions.
They should match almost 1:1
- [ ] revise